### PR TITLE
RE2022-328: fix compatibility with GFU

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: src/loaders/workspace_uploader/workspace_uploader.py [hunk 1]
```diff
@@ -1025,8 +1025,8 @@ def main():
         if proc:
             proc.terminate()
 
-    if not keep_job_dir:
-        shutil.rmtree(job_dir)
+        if not keep_job_dir:
+            shutil.rmtree(job_dir)
```
Comment (Tianhao-Gu) [inline src/loaders/workspace_uploader/workspace_uploader.py:1029]: when above in try returns, `job_dir` might not be deleted if it's outside of finally.

**Review by MrCreosote — COMMENTED** [review]
LGTM but @Xiangs18 should review

File: src/loaders/workspace_uploader/workspace_uploader.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/loaders/workspace_uploader/workspace_uploader.py:1029]: no. in try, function can return. 

```
In [4]: def example_function():
   ...:     try:
   ...:         return
   ...:     finally:
   ...:         print('foo')
   ...: 
   ...:     print('bar')
   ...: 
   ...: example_function()
foo

In [5]:
```

Comment (Xiangs18) [reply src/loaders/workspace_uploader/workspace_uploader.py:1029]: yeah, I made a toy example. 👍

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Pull request merged**